### PR TITLE
Added pagination header parsing

### PR DESCRIPTION
**Pull request by RachelTucker**
**Changes**
- Added parsing of pagination headers to response parsers (when applicable)
- Added pagination header vars to response handlers (when applicable)

--
Example Generated Code: Get Users Response Parser
--
```
/*
 * ******************************************************************************
 *   Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
 *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
 *   this file except in compliance with the License. A copy of the License is located at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file.
 *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
 *   specific language governing permissions and limitations under the License.
 * ****************************************************************************
 */

// This code is auto-generated, do not modify
package com.spectralogic.ds3client.commands.parsers;

import com.spectralogic.ds3client.commands.parsers.interfaces.AbstractResponseParser;
import com.spectralogic.ds3client.commands.parsers.utils.ResponseParserUtils;
import com.spectralogic.ds3client.commands.spectrads3.GetUsersSpectraS3Response;
import com.spectralogic.ds3client.models.SpectraUserList;
import com.spectralogic.ds3client.networking.WebResponse;
import com.spectralogic.ds3client.serializer.XmlOutput;
import com.spectralogic.ds3client.utils.ReadableByteChannelInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.nio.channels.ReadableByteChannel;

public class GetUsersSpectraS3ResponseParser extends AbstractResponseParser<GetUsersSpectraS3Response> {
    private final int[] expectedStatusCodes = new int[]{200};

    @Override
    public GetUsersSpectraS3Response parseXmlResponse(final WebResponse response, final ReadableByteChannel blockingByteChannel) throws IOException {
        final int statusCode = response.getStatusCode();
        final Integer pagingTruncated = parseIntHeader("page-truncated");
        final Integer pagingTotalResultCount = parseIntHeader("total-result-count");
        if (ResponseParserUtils.validateStatusCode(statusCode, expectedStatusCodes)) {
            switch (statusCode) {
            case 200:
                try (final InputStream inputStream = new ReadableByteChannelInputStream(blockingByteChannel)) {
                    final SpectraUserList result = XmlOutput.fromXml(inputStream, SpectraUserList.class);
                    return new GetUsersSpectraS3Response(result, pagingTotalResultCount, pagingTruncated);
                }

            default:
                assert false: "validateStatusCode should have made it impossible to reach this line";
            }
        }

        throw ResponseParserUtils.createFailedRequest(response, blockingByteChannel, expectedStatusCodes);
    }
}
```

--
Example Generated Code: Get Users Response Handler
--
```
/*
 * ******************************************************************************
 *   Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
 *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
 *   this file except in compliance with the License. A copy of the License is located at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file.
 *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
 *   specific language governing permissions and limitations under the License.
 * ****************************************************************************
 */

// This code is auto-generated, do not modify
package com.spectralogic.ds3client.commands.spectrads3;

import com.spectralogic.ds3client.models.SpectraUserList;
import com.spectralogic.ds3client.commands.interfaces.AbstractResponse;

public class GetUsersSpectraS3Response extends AbstractResponse {
    
    private final SpectraUserList spectraUserListResult;

    private final Integer pagingTotalResultCount;

    private final Integer pagingTruncated;

    public GetUsersSpectraS3Response(final SpectraUserList spectraUserListResult, final Integer pagingTotalResultCount, final Integer pagingTruncated) {
        this.spectraUserListResult = spectraUserListResult;
        this.pagingTotalResultCount = pagingTotalResultCount;
        this.pagingTruncated = pagingTruncated;
    }

    public SpectraUserList getSpectraUserListResult() {
        return this.spectraUserListResult;
    }

    public Integer getPagingTotalResultCount() {
        return this.pagingTotalResultCount;
    }

    public Integer getPagingTruncated() {
        return this.pagingTruncated;
    }

}
```

### CODE REVIEW

File: ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/responseparser/AllocateJobChunkParserGenerator.java [hunk 1]
```diff
@@ -35,7 +35,8 @@
     @Override
     public ImmutableList<ResponseCode> toResponseCodeList(
```
Comment (shabtaisharon) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/responseparser/AllocateJobChunkParserGenerator.java]: would it be better to overload this function like this, and than not to change all the place you call it and pass false for pagination?

public ImmutableList<ResponseCode> toResponseCodeList(final ImmutableList<Ds3ResponseCode> ds3ResponseCodes, final String responseName) {
	return toResponseCodeList(ds3ResponseCodes, responseName, false);
}

Comment (RachelTucker) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/responseparser/AllocateJobChunkParserGenerator.java]: There are several functions here where the parameter was added. I don't think AllocateJobChunkParserGenerator.toResponseCodeList is called often enough for this to clean up the code.  In general, this class inherits from BaseResponseParserGenerator, which does need the added parameter.

Comment (RachelTucker) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/responseparser/AllocateJobChunkParserGenerator.java]: Alternatively, I could clean up the tests by adding an alternate constructor to the ParseResponse objects, and default to no pagination headers.

Comment (shabtaisharon) [inline ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/generators/responseparser/AllocateJobChunkParserGenerator.java]: sound good :+1:

File: ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/generators/responseparser/BaseResponseParserGenerator_Test.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public void toResponseCodes_Test() {
                 new Ds3ResponseCode(400, defaultResponseType),
                 new Ds3ResponseCode(503, defaultResponseType));
 
-        final ImmutableList<ResponseCode> result = generator.toResponseCodeList(responseCodes, "TestResponse");
+        final ImmutableList<ResponseCode> result = generator.toResponseCodeList(responseCodes, "TestResponse", false);
```
Comment (shabtaisharon) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/generators/responseparser/BaseResponseParserGenerator_Test.java]: do we still need all those false parameters?

Comment (shabtaisharon) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/generators/responseparser/BaseResponseParserGenerator_Test.java]: oh sorry yes we do

**Pull request merged**